src/main/java/com/vint/shop/service/impl/ManufacturerServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.vint.shop.service.impl;
+
+import com.vint.shop.domain.Manufacturer;
+import com.vint.shop.repository.ManufacturerRepository;
+import com.vint.shop.service.ManufacturerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+
+    @Autowired
+    protected ManufacturerRepository manufacturerRepository;
+
+    @Override
+    public List<Manufacturer> findAll() {
+        return manufacturerRepository.findAll();
+    }
+
+    @Override
+    public boolean saveManufacturer(Manufacturer manufacturer) {
+        Manufacturer manufacturerFromDB = manufacturerRepository.findByName(manufacturer.getName());
+
+        if (manufacturerFromDB != null) {
```

Comment:
It's not a good idea to implement such methods this way
Save, delete, update methods SHOULDN'T contain any logic, which actually is a validation logic.
It should be performed before calling such methods